### PR TITLE
Add functionality to share Contact via QR code

### PR DESCRIPTION
**Pull request by shubhamkmr04**

### CODE REVIEW

File: components/CollapsedQR.tsx [hunk 1]
```diff
@@ -56,7 +56,7 @@ function ValueText({ value, truncateLongValue }: ValueTextProps) {
 }
 
 interface CollapsedQRProps {
-    value: string;
+    value: any;
```
Comment (kaloudis) [inline components/CollapsedQR.tsx:59]: not crazy about the loosening of types here

File: utils/handleAnything.ts [hunk 2]
```diff
@@ -364,6 +364,14 @@ const handleAnything = async (
                     localeString('utils.handleAnything.invalidLnurlParams')
                 );
             });
+    } else if (JSON.parse(data)?.id) {
```
Comment (kaloudis) [inline utils/handleAnything.ts:367]: is there a way we can better identify contacts?

Comment (shubhamkmr04) [reply utils/handleAnything.ts:367]: Do you mean by using any other property? Can't think of any other than 'id' for it as it is unique

Comment (kaloudis) [reply utils/handleAnything.ts:367]: can we turn the unique field to be `contactId`?

Comment (shubhamkmr04) [reply utils/handleAnything.ts:367]: Yeah, will do that.

**Comment by kaloudis** [comment]
Looking pretty good right now. Are there any backwards compatibility issues we created moving from `id` to `contactId`?

**Comment by shubhamkmr04** [comment]
> Looking pretty good right now. Are there any backwards compatibility issues we created moving from `id` to `contactId`?

Not in my opinion. Tested all features of contacts after this change

**Comment by kaloudis** [comment]
I found some breaking changes moving from `id` to `contactId`. You need to either add checks that moves the `id` to `contactId` on the contacts or find a different approach here. I think using a `zeuscontact:` or `lncontact:` prefix instead will be a better approach.

To see the bug, create contacts with `beta2` then switch to this branch. Then try to load a contact. It will crash trying to load the banner because the contact hasn't been loaded at all. If you want to keep those old approach you need changes on `Contacts.tsx` and `ContactDetails.tsx`.

Furthermore, the other QRs are still missing prefixes. there should be `nostr:`, `lightning:`, and `bitcoin:` prefixes on the QRs where applicable

File: utils/handleAnything.ts [hunk 3]
```diff
@@ -16,7 +16,7 @@ const isClipboardValue = (data: string) =>
     handleAnything(data, undefined, true);
 
 const handleAnything = async (
-    data: string,
+    data?: any,
```
Comment (kaloudis) [inline utils/handleAnything.ts:19]: we don't have to change the type here

**Comment by kaloudis** [comment]
Screenshot 2024-01-12 at 00 04 00

let's add more spacing between favorite and qr code here

File: views/ContactDetails.tsx [hunk 4]
```diff
@@ -556,13 +630,29 @@ export default class ContactDetails extends React.Component<
...
+                                    onPress={() => {
+                                        navigation.goBack();
+                                        navigation.navigate('AddContact', {
+                                            prefillContact: nostrContact,
+                                            isEdit: true,
+                                            isNostrContact:
+                                                this.state.isNostrContact
+                                        });
+                                    }}
+                                    title="Edit and save contact"
```
Comment (kaloudis) [inline views/ContactDetails.tsx:651]: this needs to be localized

File: views/ContactDetails.tsx [hunk 5]
```diff
@@ -234,6 +244,65 @@ export default class ContactDetails extends React.Component<
...
+                    // Add the 'zeuscontact:' prefix to the contactData parameter
+                    const zeusContactData =
+                        'zeuscontact:' +
+                        JSON.stringify(contactDataWithoutPhoto);
+
+                    navigation.navigate('ContactInfo', {
+                        contactData: zeusContactData,
+                        addressData: [
+                            ...addPrefixToAddresses(
+                                this.state.contact?.lnAddress,
```
Comment (kaloudis) [inline views/ContactDetails.tsx:279]: destructure the `contact` variable out of the state as it is used multiple times

File: views/ContactDetails.tsx [hunk 6]
```diff
@@ -556,13 +630,29 @@ export default class ContactDetails extends React.Component<
                                 )}
                         </ScrollView>
                         {this.state.isNostrContact && (
```
Comment (kaloudis) [inline views/ContactDetails.tsx:632]: destructure the `isNostrContact` variable out of the state as it is used multiple times

File: views/ContactInfo.tsx [hunk 7]
```diff
@@ -0,0 +1,167 @@
+import React, { useEffect, useState } from 'react';
```
Comment (kaloudis) [inline views/ContactInfo.tsx:1]: I don't like the name of this file `ContactInfo` can be confused with `ContactDetails` - something like `ContactQRs` would be more specific, more appropriate

However, I'm unsure if we're better off transforming the `QR` view and modifying it to handle multiple QRs than creating a new one here

File: views/ContactDetails.tsx [hunk 8]
```diff
@@ -234,6 +244,65 @@ export default class ContactDetails extends React.Component<
...
+                        contactDataWithoutPhoto.photo &&
+                        !contactDataWithoutPhoto.photo.startsWith('http')
+                    ) {
+                        delete contactDataWithoutPhoto.photo;
+                    }
+
+                    // Add the 'zeuscontact:' prefix to the contactData parameter
+                    const zeusContactData =
+                        'zeuscontact:' +
+                        JSON.stringify(contactDataWithoutPhoto);
```
Comment (kaloudis) [inline views/ContactDetails.tsx:273]: you can do this as just `zeuscontact:${contactDataWithoutPhoto}`

File: views/ContactInfo.tsx [hunk 9]
```diff
@@ -14,15 +14,15 @@ import Header from '../components/Header';
 import { themeColor } from '../utils/ThemeUtils';
```
Comment (kaloudis) [inline views/ContactInfo.tsx:14]: you can delete this file now

File: views/ContactQR.tsx [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+import { themeColor } from '../utils/ThemeUtils';
+import CollapsedQR from '../components/CollapsedQR';
+
+interface ContactQRProps {
+    navigation: any;
+}
+
+interface IntroProps {
+    navigation: any;
+}
```
Comment (kaloudis) [inline views/ContactQR.tsx:23]: you don't need this second interface

**Review by kaloudis — APPROVED** [review]

**Pull request merged**